nifi-nar-bundles/nifi-dropbox-bundle/nifi-dropbox-processors/src/main/java/org/apache/nifi/processors/dropbox/ListDropbox.java

```diff
@@ -0,0 +1,293 @@
+
+        for (DropboxFlowFileAttribute attribute : DropboxFlowFileAttribute.values()) {
+            Optional.ofNullable(attribute.getValue(entity))
+                    .ifPresent(value -> attributes.put(attribute.getName(), value));
+        }
+
+        return attributes;
+    }
+
+    @Override
+    protected String getPath(final ProcessContext context) {
+        return context.getProperty(FOLDER).evaluateAttributeExpressions().getValue();
+    }
+
+    protected DbxClientV2 getDropboxApiClient(ProcessContext context) {
+        final DropboxCredentialService credentialService = context.getProperty(CREDENTIAL_SERVICE)
+                .asControllerService(DropboxCredentialService.class);
+        DbxRequestConfig config = new DbxRequestConfig(format("%s-%s", getClass().getSimpleName(), getIdentifier()));
+        DropboxCredentialDetails credential = credentialService.getDropboxCredential();
+        return new DbxClientV2(config, new DbxCredential(credential.getAccessToken(), -1L,
+                credential.getRefreshToken(), credential.getAppKey(), credential.getAppSecret()));
+    }
+
+    @Override
+    protected List<DropboxFileInfo> performListing(ProcessContext context, Long minTimestamp,
+            ListingMode listingMode) throws IOException {
+        final List<DropboxFileInfo> listing = new ArrayList<>();
+
+        final String folderName = getPath(context);
+        final Boolean recursive = context.getProperty(RECURSIVE_SEARCH).asBoolean();
+        final Long minAge = context.getProperty(MIN_AGE).asTimePeriod(TimeUnit.MILLISECONDS);
+
+        try {
+            Predicate<FileMetadata> metadataFilter = createMetadataFilter(minTimestamp, minAge);
+
+            ListFolderBuilder listFolderBuilder = dropboxApiClient.files().listFolderBuilder(convertFolderName(folderName));
+            ListFolderResult result = listFolderBuilder
+                    .withRecursive(recursive)
+                    .start();
+
+            List<FileMetadata> metadataList = new ArrayList<>(filterMetadata(result, metadataFilter));
+
+            while (result.getHasMore()) {
+                result = dropboxApiClient.files().listFolderContinue(result.getCursor());
+                metadataList.addAll(filterMetadata(result, metadataFilter));
+            }
+
+            for (FileMetadata metadata : metadataList) {
+                DropboxFileInfo.Builder builder = new DropboxFileInfo.Builder()
+                        .id(metadata.getId())
+                        .path(getParentPath(metadata.getPathDisplay()))
+                        .name(metadata.getName())
+                        .size(metadata.getSize())
+                        .timestamp(metadata.getServerModified().getTime())
+                        .revision(metadata.getRev());
+
+                listing.add(builder.build());
+            }
+        } catch (DbxException e) {
+            throw new IOException("Failed to list Dropbox folder " + folderName, e);
```

Comment:
Recommend wrapping the folder name in brackets for easier reading:

```suggestion
            throw new IOException("Failed to list Dropbox folder [" + folderName +"]", e);
```